### PR TITLE
added tests for #348

### PR DESCRIPTION
**Pull request by ramvikrams**
added the tests for the issue #348

- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: tests/test_series.py [hunk 1]
```diff
@@ -1236,3 +1236,7 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
 
     cond3 = pd.Series([False, True, True])
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
+    
+# from issue #348 (bitwise operators on Series should support int)
+s= pd.Series([1,2,3,4])
+check(assert_type(s & 3, pd.Series), pd.Series)
```
Comment (Dr-Irv) [inline tests/test_series.py:1242]: A few comments here:
1. place the tests in a function, like the tests above it
2. add tests for the or operator ( `|` ) and xor operator (`^`)
3. add tests for the 3 reverse operators, e.g., `3 & s`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by Dr-Irv** [comment]
@ramvikrams also, be sure to run `poetry run poe style` on your code before committing so that the style checks are done.

File: tests/test_series.py (hunk 1 above)
Comment (ramvikrams) [reply tests/test_series.py:1242]: Yes sir

Comment (ramvikrams) [reply tests/test_series.py]: done with the changes sir

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
I can tell that you will fail the formatting checks, so please run `poetry run poe style`, then push,  and then I will approve the CI checks that test your changes.

File: tests/test_series.py [hunk 2]
```diff
@@ -1236,3 +1236,25 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
 
     cond3 = pd.Series([False, True, True])
     check(assert_type(s.where(cond3, other=0), "pd.Series[int]"), pd.Series, int)
+    
+def Bitwise_Operator_on_Series_Support_int() -> None:
```
[the remark this answers is not in the document]
Comment (ramvikrams) [reply tests/test_series.py:1240]: sir how do we do the poetry style checks

[the remark this answers is not in the document]
Comment (Dr-Irv) [reply tests/test_series.py:1240]: See https://github.com/pandas-dev/pandas-stubs/blob/main/docs/setup.md
and https://github.com/pandas-dev/pandas-stubs/blob/main/docs/tests.md

[the remark this answers is not in the document]
Comment (ramvikrams) [reply tests/test_series.py:1240]: sir whenever I enter the poetry shell and try to run the commands but it shows error that  **ModuleNotFoundError: No module named 'loguru'** but when I do **pip install loguru** it says that all dependencies are satisfied. So how do I proceed next

[the remark this answers is not in the document]
Comment (ramvikrams) [reply tests/test_series.py:1240]: Sir are there only two changes to make

- changing function name

- initializing s once

[the remark this answers is not in the document]
Comment (ramvikrams) [reply tests/test_series.py:1240]: sir the issue got resolved just reinstalled python

File: tests/test_series.py [hunk 3]
```diff
@@ -1236,3 +1236,25 @@ def cond2(x: pd.Series[int]) -> pd.Series[bool]:
...
+    #from issue348(bitwise operators on Series should support int)
+    s=pd.Series([1,2,34,])
+    check(assert_type(s & 3, pd.Series), pd.Series)
+    
+    s=pd.Series([1,2,34,])
```
[the remark this answers is not in the document]
Comment (ramvikrams) [reply tests/test_series.py]: Sir improved the test cases format and function name and ran the poetry run poe style and it finished off without issues

**Comment by Dr-Irv** [comment]
CI is failing with mypy for some reason, and I'll have to investigate that for you.

**Comment by ramvikrams** [comment]
Thank you sir

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
So for some reason, your changes to `core/series.pyi` are no longer in this PR.  Please make sure they are there (they were there in an earlier commit)

Secondly, you need to make the result of each of these logical operators `Series[int]`, not `Series[bool]` or `Series`.  That was an error in the previous stubs.

Thirdly, in the tests, they should be of the form:
```python
    check(assert_type(s & 3, "pd.Series[int]"), pd.Series, int)
```

Ideally, you should add tests for lists and `Series`, e.g.,:
```python
    check(assert_type(s & [4,3,2,1], "pd.Series[int]"), pd.Series, int)
    check(assert_type(s & pd.Series([4,3,2,1]), "pd.Series[int]"), pd.Series, int)
```
and do that for each of the logical operators, in the forward and reverse styles.

**Comment by ramvikrams** [comment]
Yes sir making the changes

**Comment by ramvikrams** [comment]
Sir creating a new pr and would be adding the changes in core/series.pyi and tests there because I am not able to  add the series.pyi file changes here

**Comment by ramvikrams** [comment]
done with the changes sir and ran the poetry command, it finished without an error

**Comment by Dr-Irv** [comment]
You're going to have to find a way to have both the changes to `core/series.pyi` and `tests/test_series.py` in one PR.  Choose this one or the other one.

**Comment by Dr-Irv** [comment]
xref #378

**Comment by ramvikrams** [comment]
> You're going to have to find a way to have both the changes to `core/series.pyi` and `tests/test_series.py` in one PR.  Choose this one or the other one.
> 

Yes sir will do it today

**Comment by Dr-Irv** [comment]
closing because of #378

**Pull request closed without merging**